### ISSUE TITLE
Theme manager: Suggestion for better implementation after 31135fb0

### ISSUE THREAD

**Issue by un1versal** [kvirc/KVIrc#1737]
As discussed in #1704 - (I think this should be addressed before next stable release)

https://github.com/kvirc/KVIrc/commit/31135fb0bbcd9a12b5a19990c28e5ccdcf4b06f6 was made to solve the problem that the old layout best size was screenshot at 32x32 or 64x64 or something imperceptible.

There are 2 problems with this new implementation

1) Not all existing themes have screenshots, so a default image https://github.com/kvirc/KVIrc/blob/master/data/pics/kvi_bigicon_theme.png is shown and that is now HUGE

2) Themes with screenshots Is still not ideal, looks awful and not much better than before, Im sorry to say., you need a proper screenshot size.

This is a real capture of the existing dialog; of how a theme without screenshot and a theme with screenshot look now.
![capture](https://cloud.githubusercontent.com/assets/3521959/11062728/302e4076-87a9-11e5-80b6-040f1006ebd9.PNG)

---

So clearly this is not ideal and imo more awful than before.

**Solution 1)** Revert https://github.com/kvirc/KVIrc/commit/31135fb0bbcd9a12b5a19990c28e5ccdcf4b06f6 and allow themes to have an **optional** icon + screenshots and make a live preview on mouse-over for screenshot_medium.png or screenshot png

Below is a mock for proposed solution 1 to illustrate the suggestion

![solution 1](https://cloud.githubusercontent.com/assets/3521959/11062800/f4799566-87a9-11e5-8458-976e208d7bb9.png)

**Solution 2)** Change the layout of this dialog; so the screenshot takes the whole horizontal row and the description is below the screenshot.

![solution 2](https://cloud.githubusercontent.com/assets/3521959/11062910/d6ac7b1a-87aa-11e5-8412-26190a38990d.png)

**Comment by Stanzilla** [comment]
I like 1) but not when it requires theme authors to create an icon for their theme, it's pretty hard to represent a theme as an icon so I would either have a cropped screenshot there (would probably too much work for little gain, having an algorithm to get a proper thumbnail out of it) or use a generic icon, like it did before, when no custom icon can be found

**Comment by un1versal** [comment]
An icon can be any image anyone likes, but if none is included the default is used, I see no issue here.

**Comment by Stanzilla** [comment]
> The icon itself is a quickly comprehensible symbol of a software tool, function, or a data file, accessible on the system and is more like a traffic sign than a detailed illustration of the actual entity it represents.
> (https://en.wikipedia.org/wiki/Icon_(computing))

An icon should not be a screenshot. That's why I said it's hard to represent a theme with an icon and you'd have to come up with something extra, not a screenshot, and I can see theme authors (there are none active btw (apart from you), at least I haven't heard from one in ages) not wanting to do that, so a unique placeholder icon for all themes that don't come with an icon would be good. Also it needs to work with every old theme that is not maintained and the author gone, which is another reason a generic approach would be better than requiring user/author action.

Other companies (Google/Microsoft) do this with a square and glyphs that represent the name of the entry, for example contacts have a square with an `A` if the contact `Arnold` has no contact picture. This would fit quite well here, too and you even started doing that with your own themes, so I like that.

The problem here is, that before you started doing that, the image (generic term here) next to the description of the theme was a screenshot thumbnail, _not_ an icon, which is important for the decision as to what goes there now because _now_ we have a mix of thumbnails and icons.

**Comment by un1versal** [comment]
Again, these are suggestions, to improving a real issue, please dont [bikeshed](http://bikeshed.com/)

**Comment by Stanzilla** [comment]
@un1versal I have no idea what that means.

**Comment by un1versal** [comment]
> @un1versal I have no idea what that means.

Read [bikeshed](http://bikeshed.com/) You're nitpicking on the icon when in fact if a theme has no screenshot it always used https://github.com/kvirc/KVIrc/blob/master/data/pics/kvi_bigicon_theme.png so its neither here nor there for the grand scheme of things. Your disagreement is noted, but since I dont see any themes by stanzilla your objection to me is bikeshedding for the sake of it.

**Comment by Stanzilla** [comment]
I'm not nitpicking, I just explained what the problem is.

If we already have a placeholder icon for when no icon is present it should always be shown instead of a screenshot, that's what I was going for.

> but since I dont see any themes by stanzilla 
> http://kvirc.net/?id=themes&lang=en
> Minimalist 1.0

**Comment by un1versal** [comment]
Still see no problem, this would be optional, not mandatory since there is already a placeholder which atm is only a placeholder for when no screenshot is present so if it was a placeholder for absent icon you see no difference anyway. it would still look like it does now and like it has always looked.

However for themes made now, its mandatory to have a screenshot,

as a side offtopic note: a real issue is that all old themes with icons will very likely be broken and or incomplete and themes that include texticons and other files will be even more incompatible and broken.

Progress, cant do anything about that, something has to give and it has given.

**Comment by un1versal** [comment]
I had a chance to discuss this with pragma and we agreed that a 3rd option mixing 1/2 would be even better and more ideal.

**Solution 3)**

![53qqsui4](https://cloud.githubusercontent.com/assets/3521959/11166919/3b5a00fe-8b44-11e5-8158-ab77226a9785.png)

1st entry illustrates when themes have no screenshot but has theme icon

2nd entry illustrates when themes have no screenshots and no theme icon (no change from what is done currently)

3rd entry illustrates when themes have screenshot + theme icon.

**Comment by Stanzilla** [comment]
:+1: Just needs a higher res  version of the placeholder icon, but I bet you had that already planned.

**Comment by un1versal** [comment]
Yes, I have something much nicer in mind for the placeholder icon. But its early days yet for any show and tell.

**Comment by un1versal** [comment]
Still a long long way to go... but https://github.com/kvirc/KVIrc/compare/master...un1versal:Mary-had-a-little-lamb

That makes the dialog look like
![capture](https://cloud.githubusercontent.com/assets/3521959/11396957/9785aaaa-936d-11e5-8762-48aa2633ecfc.PNG)

Im missing the screenshot and icon parts which atm ive no idea how to do. utterly lost there.

**Comment by un1versal** [comment]
as above I reverted in https://github.com/kvirc/KVIrc/commit/621e47d890d71220fc7c2cc463e06c652b4badf0

Until @pragmaware or someone designs the agreed upon with pragmaware dialog in https://github.com/kvirc/KVIrc/issues/1737#issuecomment-156774237 best not have huge pixelated and useless screenshot sizes that are no more helpful than it originally was.

I have the default theme icon on the forge, when someone adds support please ping so I can finish and push the design up.
